### ISSUE TITLE
Don't emit prefer-w when both `i` and `u` flags are set

### ISSUE THREAD

**Issue by tats-u** [ota-meshi/eslint-plugin-regexp#811]
**Information:**
- ESLint version:
- `eslint-plugin-regexp` version:

**Description**

When both `i` and `u` flags are set, `\w` matches U+017F & U+212A, too.

https://github.com/eslint/eslint/issues/19474#issuecomment-2698133954

```js
const ex1 = /^\w+$/i;
const ex2 = /^\w+$/iu;

const str = "\u017f\u212a";

// ESLint expected: true / actual: false
console.log(ex1.test(str) === ex2.test(str));
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Character_class_escape#w

> If the regex is [Unicode-aware](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/unicode#unicode-aware_mode) and the [i](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/ignoreCase) flag is set, it also matches other Unicode characters that get canonicalized to one of the characters above through [case folding](https://unicode.org/Public/UCD/latest/ucd/CaseFolding.txt).

↑This makes U+017F and U+212A matches `\w` in `iu` flag.
